pyspark_ai/ai_utils.py

```diff
@@ -132,3 +132,17 @@ def extract_code_blocks(text: str) -> List[str]:
             if text.startswith("`") and text.endswith("`"):
                 return [text.strip("`")]
             return [text]
+
+    @staticmethod
+    def retry_execution(callback, max_tries=3):
+        tries = 0
+        last_exception = None
+        while max_tries > tries:
+            try:
+                return callback()
```

Comment:
We need to have a conversation with LLM and feedback the error message

Reply:
Adjusted in [8ef79c0](https://github.com/pyspark-ai/pyspark-ai/pull/132/commits/8ef79c0b8c6789962a41770f3d1d08ce2d1d5ba6)

Reply:
@xinrong-meng could you verify the feature with gpt-3.5 or other models?